massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -1,173 +1,193 @@
-            max_final_events: 200,
-        },
-        thread_count: 2,
-        genesis_timestamp: MassaTime::now().unwrap(),
-        t0: 16000.into(),
-        clock_compensation: 0,
-    };
-    (initial_file, res)
+/// Get a randomized address
+pub fn get_random_address() -> Address {
+    get_random_address_full().0
 }
 
-fn get_sample_ledger() -> SCELedger {
-    SCELedger(
-        vec![
-            (
-                get_random_address(),
-                SCELedgerEntry {
-                    balance: Amount::from_str("129").unwrap(),
-                    opt_module: None,
-                    data: vec![
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testA".as_bytes()),
-                            "test1_data".into(),
-                        ),
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testB".as_bytes()),
-                            "test2_data".into(),
-                        ),
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testC".as_bytes()),
-                            "test3_data".into(),
-                        ),
-                    ]
-                    .into_iter()
-                    .collect(),
-                },
-            ),
-            (
-                get_random_address(),
-                SCELedgerEntry {
-                    balance: Amount::from_str("878").unwrap(),
-                    opt_module: Some("bytecodebytecode".into()),
-                    data: vec![
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testD".as_bytes()),
-                            "test4_data".into(),
-                        ),
-                        (
-                            massa_hash::hash::Hash::compute_from("key_testE".as_bytes()),
-                            "test5_data".into(),
-                        ),
-                    ]
-                    .into_iter()
-                    .collect(),
-                },
-            ),
-        ]
-        .into_iter()
-        .collect(),
-    )
+fn get_sample_ledger() -> Result<(Arc<RwLock<FinalLedger>>, NamedTempFile), LedgerError> {
+    let mut initial: BTreeMap<Address, Amount> = Default::default();
+    initial.insert(get_random_address(), Amount::from_str("129").unwrap());
+    initial.insert(get_random_address(), Amount::from_str("878").unwrap());
+    let (cfg, tempfile) = LedgerConfig::sample(&initial);
+    Ok((Arc::new(RwLock::new(FinalLedger::new(cfg)?)), tempfile))
 }
 
-#[tokio::test]
+#[test]
 #[serial]
-async fn test_execution_basic() {
-    let (_config_file_keepalive, settings) = get_sample_settings();
-    assert!(start_controller(settings, None).await.is_ok());
+fn test_execution_basic() {
+    let (sample_ledger, _keep) = get_sample_ledger().unwrap();
+    let (_, _) = start_execution_worker(ExecutionConfig::default(), sample_ledger);
```

Comment:
Maybe you can close the worker too ? (but that makes this test a duplicate of test_execution_shutdown)

Reply:
Yes it's a little bit duplicated but it takes no time. Also we can easily understand if it's the `stop` or the `run` who fail with that one.